modules/stochastic_tools/src/state_sim/model/actions/StateSimActTransition.C

```diff
@@ -0,0 +1,167 @@
+{
+  if (prob < 0) //use the remainder so put it on the back of the lise.
+  {
+    _to_state_probs.push_back(prob);
+    _to_state_var_probs.push_back(NULL);
+    _to_states.push_back(&to_state);
+    _fail_desc.push_back(fail_desc);
+  }
+  else
+  {
+    _to_state_probs.insert(_to_state_probs.begin(), prob);
+    _to_state_var_probs.insert(_to_state_var_probs.begin(), NULL);
+    _to_states.insert(_to_states.begin(), &to_state);
+    _fail_desc.insert(_fail_desc.begin(), fail_desc);
+  }
+}
+
+void
+StateSimActTransition::removeToState(StateSimState & rem_state)
+{
+  StateSimState * a = &rem_state;
+  //auto func = [&a](const StateSimState * b) -> bool {return a==b;};
+  auto found_state = std::find_if(_to_states.begin(), _to_states.end(), [&a](const StateSimState * b) { return a == b; });
+  if (found_state != _to_states.end())
+  {
+    int idx = found_state - _to_states.begin();
+    _to_states.erase(found_state);
+    _to_state_probs.erase(_to_state_probs.begin() + idx);
+    _to_state_var_probs.erase(_to_state_var_probs.begin() + idx);
+    _fail_desc.erase(_fail_desc.begin() + idx);
+  }
+}
+
+std::vector<std::shared_ptr<StateSimIdAndDesc>>
+StateSimActTransition::WhichToState()
+{
+  std::vector<std::shared_ptr<StateSimIdAndDesc>> ret_list;
+
+  if (_has_var_probs)
+  {
+    for (unsigned int i = 0; i < _to_state_var_probs.size(); i++)
+    {
+      if (_to_state_var_probs[i] != NULL)
+      {
+        _to_state_probs[i] = ((StateSimVariable *)_to_state_var_probs[i])->getDouble();
+      }
+    }
+  }
+
+  double prob_sum = 0;
+  for (auto & n : _to_state_probs)
+  {
+    prob_sum += n;
+  }
+
+  if (_mut_excl && (prob_sum > 0) && (prob_sum < 1.0))
+    mooseAssert(false, "No default state entered for " + this->name() + " and transitions probabilites don't add up to 1.0.");
+  //throw std::runtime_error("No default state entered for " + this->name() + " and transitions probabilites don't add up to 1.0.");
+  else if (_to_states.size() == 0) //Not sure if this is an error what if we want to just leave the state.
+    mooseAssert(false, "No to states for transition " + this->name());
+  //throw std::runtime_error("No to states for transition " + this->name());
+  else if ((_to_state_probs[0] == 1.0) || (_to_state_probs[0] == -1.0))
+  {
+    //std::make_shared<StateSimEvFailRate>("A", 0.1, TimespanH(1), TimespanH(24))
+    ret_list.push_back(std::make_shared<StateSimIdAndDesc>(_to_states[0]->id(), _fail_desc[0]));
+    return ret_list;
+  }
+
+  double rand_num = nextRandom();
+  double sum = 0.0;
+  std::unordered_map<int, int> added;
+  for (unsigned int slot = 0; slot < _to_state_probs.size(); slot++)
+  {
+    int id = _to_states[slot]->id();
+
+    if (_mut_excl) //use the prev bound is the low
+    {
+      if (rand_num <= _to_state_probs[slot] + sum)
+      {
+        ret_list.push_back(std::make_shared<StateSimIdAndDesc>(id, _fail_desc[slot]));
```

Comment:
I think @permcody  was suggesting you do `ret_list.emplace_back(std::make_shared<StateSimIdAndDesc>(id, _fail_desc[slot]));`